chromium_src/chrome/browser/ui/views/tabs/tab_drag_controller.cc

```diff
@@ -0,0 +1,63 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "chrome/browser/ui/views/tabs/tab_drag_controller.h"
+
+#define TabDragController TabDragControllerChromium
+#include "src/chrome/browser/ui/views/tabs/tab_drag_controller.cc"
+#undef TabDragController
+
+#include "brave/browser/ui/views/tabs/features.h"
+
+TabDragController::~TabDragController() = default;
+
+// In order to use inner class defined in tab_drag_controller.cc, this function
+// is defined here.
+void TabDragController::Init(TabDragContext* source_context,
+                             TabSlotView* source_view,
+                             const std::vector<TabSlotView*>& dragging_views,
+                             const gfx::Point& mouse_offset,
+                             int source_view_offset,
+                             ui::ListSelectionModel initial_selection_model,
+                             EventSource event_source) {
+  TabDragControllerChromium::Init(source_context, source_view, dragging_views,
+                                  mouse_offset, source_view_offset,
+                                  initial_selection_model, event_source);
+  if (!tabs::features::ShouldShowVerticalTabs())
+    return;
+
+  // Adjust coordinate for vertical mode.
+  source_view_offset = mouse_offset.y();
+  start_point_in_screen_ = gfx::Point(mouse_offset.x(), source_view_offset);
+  views::View::ConvertPointToScreen(source_view, &start_point_in_screen_);
+
+  last_point_in_screen_ = start_point_in_screen_;
+  last_move_screen_loc_ = start_point_in_screen_.y();
+}
+
+gfx::Point TabDragController::GetAttachedDragPoint(
+    const gfx::Point& point_in_screen) {
+  if (!tabs::features::ShouldShowVerticalTabs())
```

Comment:
this doesn't look right, shouldn't you be calling the superclass method if ShouldShowVerticalTabs is not enabled?